### DISCUSSION TITLE
Set number of months as range vs. RangeStart and RangeEnd

### DISCUSSION THREAD

**Discussion by lundejd** [microsoft/finops-toolkit#965] [Q&A] [answered]
We have a number of reports that can break easily by pulling too much data. When you set the RangeStart, the data grows every month eventually breaking. With Cost Management, you could just tell it the number of months to collect and you could leave it run without having to continuously modify the start date.

We have some reports where we pull as much historical data as we can without breaking the report, which means I have to go in constantly and edit the RangeStart setting.

The setting now is: RawData = ftk_Hub("focus cost", RangeStart, RangeEnd),

Is there a way I can just set number of months? I see you can manage the parameter, but it doesn't like anything but time.

**Comment by flanakin** [comment, marked as answer]
@lundejd Sorry for missing this. I can give you the backstory if you're interested, but the short answer is we re-introduced the `Number of Months` parameter in either 0.4 or 0.5. I don't recall which (sorry!). But it is back and I do agree it works better than RangeStart/End. RangeStart/End are really only there to support incremental refresh.

Thanks for bringing this idea! Sorry for not spotting it sooner.
